src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Metrics/RuntimeMetrics.cs

```diff
@@ -33,6 +38,8 @@ static RuntimeMetrics()
             };
         }
 
+#pragma warning disable CA1823 // suppress unused fields warning, as the fields are used to keep the meters alive
```

Comment:
@noahfalk @tarekgh Is it actually necessary to keep the meters in static fields that are otherwise unused?

Reply:
If the field is otherwise unused it isn't necessary to have it. When Meter.CreateXYZ instrument is called the instrument reference is stored in the Meter's _instruments collection (https://source.dot.net/#System.Diagnostics.DiagnosticSource/System/Diagnostics/Metrics/Meter.cs,587) and it will remain rooted as long as the Meter itself is rooted and not yet Dispose()ed

Reply:
Deleted the unnecessary static fields